File: leetcode/algorithms/linked-list/reverse-linked-list-ITERATIVE.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Reverse Linked List"
*/

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode trailer = null,current=head;
        while(current!=null){
            ListNode nextNode = current.next;
            current.next=trailer;
            trailer=current;
            current=nextNode;
        }
        return trailer;
    }
}

File: leetcode/algorithms/linked-list/reverse-linked-list-ITERATIVE.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        
        prev = None;
        curr = head;
        nxt = None;
        
        while(curr!=None):
            nxt = curr.next;
            curr.next = prev;
            prev = curr;
            curr = nxt;
        
        return prev;
        